core/src/main/java/org/apache/calcite/sql/SqlSelect.java

```diff
@@ -171,6 +190,14 @@ public void setHaving(SqlNode having) {
     this.having = having;
   }
 
+  public final SqlNode getQualify() {
+    return qualify;
+  }
+
+  public void setQualify(SqlNode qualify) {
```

Comment:
Is there a situation where the qualify field would be changed after the SqlSelect object is constructed?

Reply:
I'm not sure, I added it since the other parameters also have getters and setters. But if it's not needed I can remove it

Reply:
Agree with Daniel. We can remove this setter if there aren't any users today to make it clear that it should be initialized in the constructor.

Reply:
Cool SGTM